monai/transforms/signal/array.py

```diff
@@ -0,0 +1,773 @@
+    """
+    Add a random sinusoidal signal to the input signal
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self, boundaries: Optional[List[float]] = None, frequencies: Optional[List[float]] = None, *args, **kwargs
+    ) -> None:
+        """
+        Args:
+            boundaries: list defining lower and upper boundaries for the sinusoidal magnitude,
+            lower and upper values need to be positive example : ``[0.2, 0.6]``
+            frequencies: list defining lower and upper frequencies for sinusoidal
+            signal generation example : ``[0.001, 0.02]``
+        """
+        super().__init__()
+        if boundaries is None or None in boundaries:
+            raise ValueError("Incompatible values: boundaries needs to be a list of float.")
+        if (boundaries is None) or (None in boundaries) or (any(x < 0 for x in boundaries)):
+            raise ValueError("Incompatible values: boundaries needs to be a list of positive float.")
+
+        self.boundaries = boundaries
+        self.frequencies = frequencies
+
+    def _randomize(self):
+        super().randomize(None)
+        self.magnitude = self.R.uniform(low=self.boundaries[0], high=self.boundaries[1])
+        self.freqs = self.R.uniform(low=self.frequencies[0], high=self.frequencies[1])
+
+    def __call__(self, signal: Any) -> Any:
+        """
+        Args:
+            signal: input 1 dimension signal to which sinusoidal signal will be added
+        """
+        self._randomize()
+        if len(signal.shape) == 1:
+            signal = np.expand_dims(signal, axis=0)
+        length = signal.shape[1]
+
+        time = np.arange(0, length, 1)
+        sine = self.magnitude * np.sin(self.freqs * time)
+
+        return signal + sine
+
+
+class SignalRandAddSquarePulse(RandomizableTransform):
+    """
+    Add a random square pulse signal to the input signal
+    """
+
+    backend = [TransformBackends.NUMPY]
+
+    def __init__(
+        self, boundaries: Optional[List[float]] = None, frequencies: Optional[List[float]] = None, *args, **kwargs
+    ) -> None:
+        """
+        Args:
+            boundaries: list defining lower and upper boundaries for the square pulse magnitude,
+            lower and upper values need to be positive example : ``[0.2, 0.6]``
+            frequencies: list defining lower and upper frequencies for the square pulse
+            signal generation example : ``[0.001, 0.02]``
+        """
+        super().__init__()
+        if boundaries is None or None in boundaries:
+            raise ValueError("Incompatible values: boundaries needs to be a list of float.")
+        if (boundaries is None) or (None in boundaries) or (any(x < 0 for x in boundaries)):
+            raise ValueError("Incompatible values: boundaries needs to be a list of positive float.")
+
+        self.boundaries = boundaries
+        self.frequencies = frequencies
+
+    def _randomize(self):
```

Comment:
Maybe change to override `randomize()` directly?
Same for all the other transforms.

Thanks.

Reply:
used self.randomize() directly